message/event/uploadEvent.go

```diff
@@ -0,0 +1,42 @@
+package event
+
+import (
+	"encoding/json"
+	"net/url"
+	"strings"
+)
+
+type UploadEvent struct {
+	Time  int64  `json:"time"`
+	S3URL string `json:"s3URL"`
+}
+
+func (event *UploadEvent) ConvertToEventDetails() (*EventDetails, error) {
```

Comment:
The kafka message is converted to the struct type with json.Marshall - as a result it won't convert the S3URL to a URL - having it as a URL and not a string is required to extract some of the other fields. I added this convert so the model that gets passed around is in correct state.